onnxruntime/core/providers/qnn/builder/opbuilder/batch_norm_op_builder.cc

```diff
@@ -1,26 +1,488 @@
+
+  Status CheckMinMax(float& rmin, float& rmax) const {
+    // Ensure a minimum range of 0.0001 (required by QNN)
+    rmax = std::max(rmax, rmin + 0.0001f);
+
+    // Both QNN and ORT require the range to include 0.0f
+    rmin = std::min(rmin, 0.0f);
+    rmax = std::max(rmax, 0.0f);
+
+    return Status::OK();
+  }
+
+  Status GetQminQmax(const Qnn_DataType_t qnn_data_type,
+                     float& qmin,
+                     float& qmax) const {
+    // onnx only supports 8 bits quantization
+    if (qnn_data_type == QNN_DATATYPE_SFIXED_POINT_8) {
+      qmin = static_cast<float>(std::numeric_limits<int8_t>::min());
+      qmax = static_cast<float>(std::numeric_limits<int8_t>::max());
+    } else if (qnn_data_type == QNN_DATATYPE_UFIXED_POINT_8) {
+      qmin = static_cast<float>(std::numeric_limits<uint8_t>::min());
+      qmax = static_cast<float>(std::numeric_limits<uint8_t>::max());
+    } else {
+      ORT_RETURN_IF(true, "Qnn Data Type: %d not supported yet.", qnn_data_type);
+    }
+    return Status::OK();
+  }
+
+  Status GetQminQmax(const Qnn_DataType_t qnn_data_type,
+                     int& qmin,
+                     int& qmax) const {
+    // onnx only supports 8 bits quantization
+    if (qnn_data_type == QNN_DATATYPE_SFIXED_POINT_8) {
+      qmin = static_cast<int>(std::numeric_limits<int8_t>::min());
+      qmax = static_cast<int>(std::numeric_limits<int8_t>::max());
+    } else if (qnn_data_type == QNN_DATATYPE_UFIXED_POINT_8) {
+      qmin = static_cast<int>(std::numeric_limits<uint8_t>::min());
+      qmax = static_cast<int>(std::numeric_limits<uint8_t>::max());
+    } else {
+      ORT_RETURN_IF(true, "Qnn Data Type: %d not supported yet.", qnn_data_type);
+    }
+    return Status::OK();
+  }
+
+  Status GetQuantParams(float rmin,
+                        float rmax,
+                        const Qnn_DataType_t qnn_data_type,
+                        float& scale,
+                        int& zero_point) const {
+    ORT_RETURN_IF_ERROR(CheckMinMax(rmin, rmax));
+    float qmin = 0.0f;
+    float qmax = 255.0f;
+    ORT_RETURN_IF_ERROR(GetQminQmax(qnn_data_type, qmin, qmax));
+
+    scale = (rmax - rmin) / (qmax - qmin);
+    const float initial_zero_point = qmin - (rmin / scale);
+    zero_point = static_cast<int>(RoundHalfToEven(saturate(qmax, qmin, initial_zero_point)));
+    // To match QNN quantization definition
+    zero_point = 0 - zero_point;
+    return Status::OK();
+  }
+
+  inline Status GetValueOnQnnDataType(const Qnn_DataType_t qnn_data_type,
+                                      const uint8_t* raw_ptr,
+                                      double& value,
+                                      int& offset) const {
+    switch (qnn_data_type) {
+      case QNN_DATATYPE_INT_8:
+      case QNN_DATATYPE_SFIXED_POINT_8: {
+        value = static_cast<double>(*const_cast<int8_t*>(reinterpret_cast<const int8_t*>(raw_ptr)));
```

Comment:
Doesn't seem like the `const_cast` should be necessary. I believe the following should also work:
```C++
value = static_cast<double>(*reinterpret_cast<const int8_t*>(raw_ptr));
```